ton_client/src/abi/internal.rs

```diff
@@ -86,3 +86,51 @@ pub(crate) fn create_tvc_image(
 
     Ok(image)
 }
+
+/// Determines, if public key consists only zeroes, i.e. is empty.
+pub(crate) fn is_empty_pubkey(pubkey: &ed25519_dalek::PublicKey) -> bool {
+    for b in pubkey.as_bytes() {
+        if *b != 0 {
+            return false;
+        }
+    }
+    true
+}
+
+/// Resolves public key from deploy set, tvc or signer, using this priority:
+/// 1. Initial public key from the deploy set
+/// 2. Public key from TVC image
```

Comment:
Add test calling `encode_message` with all listed options. Always add full test, because unit test could be not enough if you, for example, just forgot to call the function in proper place